internal/service/config.go

```diff
@@ -1,15 +1,30 @@
 package service
 
-import "github.com/modprox/libmodprox/configutil"
+import (
+	"fmt"
+
+	"github.com/modprox/libmodprox/configutil"
+)
 
 type Configuration struct {
-	Index PersistentStore `json:"persistent_index"`
+	// DevMode indicates if the server is in local development mode. Defaults to false.
+	DevMode     bool            `json:"dev_mode"`
+	CSRFAuthKey string          `json:"csrf_auth_key"`
+	Index       PersistentStore `json:"persistent_index"`
 }
 
 func (c Configuration) String() string {
 	return configutil.Format(c)
 }
 
+// MustCSRFAuthKey returns a valid 32-byte slice from configuration or panics.
+func (c Configuration) MustCSRFAuthKey() []byte {
```

Comment:
definitely fine for now, in the long term i'd like to handle configuration errors gracefully - providing log statements with examples or anchored links into the documentation regarding bad or missing config values